docs/DeveloperGuide.md

```diff
@@ -4,26 +4,167 @@
+3. Then, `InvMgr` will continuously loop, doing the following:
+   i. `InvMgr` will call upon `Ui` to get input from the user.
+   ii. `InvMgr` passes the user input to `Parser.parse(command)`.
+   iii. `Parser.parse(command)` returns a `Command` object.
+   iv. `InvMgr` calls upon the `execute()` method of the returned `Command` object
+4. The loop stops when the user types `exit`.
+
+### UI Component
+![UiClassDiagram](img/UiClassDiagram.png)
+
+The `UI` component consists of a single `Ui` class.
+
+The `UI` component
+- Displays salutations, prompts for user input, error messages and results of queries.
+- Reads in user inputs
+- Depends on the `Messages` and `InvMgrException` classes in the `Common` component. It displays messages stored in the `Messages` class and displays an error message whenever `InvMgrException` is invoked.
+
+### Parser Component
+
+![ParserClassDiagram](img/ParserClassDiagram.png)
+
+The diagram above shows the class diagram of how the `Parser` component works.
+
+1. The `parse(userInput)` method in `Parser` is called from `InvMgr`, and takes in the raw user input. 
+2. The method checks through a logic gate whether the user input is valid, and returns 
+a `Command` class based on the user input.
+
+### Command Component
+![CommandClassDiagram](img/CommandClassDiagram.png)
+The following diagram shows the class diagram for `Command`.
+
+`Command` is an abstract class that sets certain commonalities that is implemented across all types of commands - `AddCommand`, `DescCommand`, `ListCommand`, `DeleteCommand`, `HelpCommand`, `ExitCommand`. Each of these classes have to override the `Command`'s `execute()` method as each command has a different execution. For example, `AddCommand` will be focused on adding an item to an inventory list whereas `DescCommand` will be about retrieving information from the inventory list.
+
+### Storage Component
+
+The following diagram shows the class diagram for `Storage`.
+
+![StorageClassDiagram](img/StorageClassDiagram.png)
+
+1. `Storage` has `save()` and `load()` methods. These are called by `InvMgr` when needed.
+2. `save(itemList)` writes the contents of an `itemList` to a file.
+3. `load(itemList)` loads a JSON file into an `itemList`.
+4. Gson is used to serialise and deserialise the `itemList` of type `ArrayList<Item>`.
+
+## Implementation
+
+### Description Command
+![DescCommandSequenceDiagram](img/DescCommandSequenceDiagram.png)
+The following diagram shows the sequence diagram for retrieving the description of an item.
+
+For a user who is unaware of what an item is about, he/she can enter the command eg. `desc 2` command to extract the description for the second item in the inventory list. This command is interpreted by the `Parser` and a `DescCommand` is returned to `InvMgr`. `InvMgr` calls the execute command of `DescCommand` which retrieves the item's information from the `ItemList` and then outputs them into the `Ui` for the user to see.
```

Comment:
![image](https://user-images.githubusercontent.com/69455499/160962868-8797d01f-9d47-48e9-9263-737eaf8f1dfc.png)
In the UML diagram, the program calls writeData(ArrayList) at the end after the execution of the desc command. Not sure if this was intended as this was not explained the running of the desc command.